### PR TITLE
8492 New Dashboard data integration

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:*
https://github.com/bcgov/entity/issues/#8492

*Description of changes:*
* Integrate NR Data into new dashboard
* Handle varying data display outputs for each column dependent on affiliation type and state.
* FilingTypes enabled lookup from namex for Incorporations.
* Updated jest config and ts related packages.

*Notes:* 
* Still some sonar duplication until the old dashboard is removed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Review by karthik-aot — APPROVED** [review]

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 1]
```diff
@@ -104,71 +136,217 @@ import Vue from 'vue'
...
+  /** Returns the folio number or a default message */
+  private folio (item: Business): string {
+    return item.folioNumber || this.$t('notAvailable').toString()
+  }
+
+  /** Returns the last modified date string or a default message */
+  private lastModified (item: Business): string {
+    return item.lastModified
+      ? new Date(item.lastModified).toLocaleDateString('en-CA', {
+        timeZone: 'America/Vancouver'
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:262]: will it be good to use with moment.js?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 2]
```diff
@@ -104,71 +136,217 @@ import Vue from 'vue'
...
+      })
+      : this.$t('notAvailable').toString()
+  }
+
+  /** Returns the modified by value or a default message */
+  private modifiedBy (item: Business): string {
+    switch (true) {
```
Comment (shabeeb-aot) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:269]: may be if else will be better than switch in this case

**Review by shabeeb-aot — APPROVED** [review]

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 3]
```diff
@@ -5,24 +5,36 @@
...
+            <td class="header" v-else>
+              <strong>{{ name(item) }}</strong>
+            </td>
+            <td v-if="showCol(headers[1].text)">{{ number(item) }}</td>
+            <td v-if="showCol(headers[2].text)" class="type-col">
+              <span class="header"><strong>{{ type(item) }}</strong></span><br>
+              <span v-if="isNameRequest(item.corpType.code)">{{ typeDescription(item.nameRequest.legalType) }}</span>
+            </td>
+            <td v-if="showCol(headers[3].text)">{{ status(item) }}</td>
+            <td v-if="showCol(headers[4].text) && isPremiumAccount">{{ folio(item) }}</td>
```
Comment (sumesh-aot) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:34]: Is folio displayed only for premium account in new UX? Folio is something which can be present regardless of the account type.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue [hunk 4]
```diff
@@ -9,96 +9,120 @@
     </div>
     <v-container class="view-container">
       <div class="view-header align-center">
-        <h1 class="view-header__title">Manage Businesses</h1>
+        <h1 class="view-header__title">My Business Registry<br>
+          <span class="subtitle">Start BC-based businesses and keep business records up to date.</span>
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:13]: Updated in next commit

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 1 above)
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:262]: Will look into this 👍

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 2 above)
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:269]: For sure, can do 👍

**Review by sumesh-aot — APPROVED** [review]

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 2 above)
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:269]: updated in next commit

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 3 above)
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:34]: Currently yes, in the requirements/designs it was called out as for premium accounts only. 
Should i bring this up with our design team?

Screen Shot 2021-08-30 at 10 17 36 AM

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue (hunk 1 above)
Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:262]: Updated in next commit 👍

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: Wait! Moment is obsolete. I recommend you look it up and updat this code accordingly (may want to create your own date mixin).

Also, JS time/date functions assume the user's browser/computer is set to the correct time/date, which can be faked (or just wrong). If time/date is used for decision-making then you shouldn't trust the user's time/date. This is a larger discussion - bring it up if necessary. I suspect, here, time/date is display only.

Comment (cameron-freshworks) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: This is for display only, you bet! I'm not sure this one display merits it's own Mixin in this ui though. 
I think i prefer to set it back to it how it was originally, i didn't realize Moment is no longer under development, although i'm not sure it's going anywhere anytime soon. 
Thanks for the heads up here!

**Pull request merged**